configure.ac

```diff
@@ -433,9 +433,12 @@ fi
 
 AC_CACHE_CHECK([for C99 vsnprintf],rsync_cv_HAVE_C99_VSNPRINTF,[
 AC_TRY_RUN([
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include <sys/types.h>
 #include <stdarg.h>
-void foo(const char *format, ...) {
+static void foo(const char *format, ...) {
```

Comment:
This change especially looks wrong. While supporting old compilers is not really important, `static` is a new thing and completely unnecessary here.

Reply:
It isn't 100% needed to fix the problem, but I was just trying to make the code fragment compile warning-free to make it as reliable as possible.  Is there any reason that you think it shouldn't be marked `static`?  It clearly isn't an exported function.

Reply:
because `static` is a C99 keyword, and it makes the test no long C89. While we don't support C89 that defeats the whole purpose of what autotools is.

Reply:
That is just not true.  `static` existed long before C99, or even C89.  It was in use even in the original Bell Labs compilers in the 1970s.

Here's an example from Dennis Richie himself: https://github.com/dspinellis/unix-history-repo/blob/Research-V7-Snapshot-Development/usr/src/libc/gen/ttyslot.c#L40

It predates the `<stdarg.h>` header by at least a decade, so there is no risk that a compiler would be old enough to pass this test and not support something as basic as `static`.  Indeed, a UNIX machine that old wouldn't even be able to run `./configure` since the Bourne shell wouldn't be featureful enough.